qt/bundle/win/anki.template.nsi

```diff
@@ -127,10 +130,43 @@ FunctionEnd
 
 ;--------------------------------
 
+; Macro from fileassoc changed to work non elevated
+!macro APP_ASSOCIATE_HKCU EXT FILECLASS DESCRIPTION ICON COMMANDTEXT COMMAND
+  ; Backup the previously associated file class
+  ReadRegStr $R0 HKCU "Software\Classes\.${EXT}" ""
+  WriteRegStr HKCU "Software\Classes\.${EXT}" "${FILECLASS}_backup" "$R0"
+ 
+  WriteRegStr HKCU "Software\Classes\.${EXT}" "" "${FILECLASS}"
+ 
+  WriteRegStr HKCU "Software\Classes\${FILECLASS}" "" `${DESCRIPTION}`
+  WriteRegStr HKCU "Software\Classes\${FILECLASS}\DefaultIcon" "" `${ICON}`
+  WriteRegStr HKCU "Software\Classes\${FILECLASS}\shell" "" "open"
+  WriteRegStr HKCU "Software\Classes\${FILECLASS}\shell\open" "" `${COMMANDTEXT}`
+  WriteRegStr HKCU "Software\Classes\${FILECLASS}\shell\open\command" "" `${COMMAND}`
+!macroend
+
+; Macro from fileassoc changed to work non elevated
+!macro APP_UNASSOCIATE_HKCU EXT FILECLASS
+  ; Backup the previously associated file class
+  ReadRegStr $R0 HKCU "Software\Classes\.${EXT}" `${FILECLASS}_backup`
+  WriteRegStr HKCU "Software\Classes\.${EXT}" "" "$R0"
+ 
+  DeleteRegKey HKCU `Software\Classes\${FILECLASS}`
+!macroend
+
 ; The stuff to install
 Section ""
 
-  SetShellVarContext all
+  SetShellVarContext current
+
+  ; "Upgrade" from elevated anki
+  ReadRegStr $0 HKLM "Software\WOW6432Node\Anki" "Install_Dir64"
+  ${IF} $0 != ""
+      ; old value exists, we want to call uninstall with silent parameter
+      ; the wait does not work, I suspect this is because of the UAC prompt
+      ; it is still better to use the wait version in my opinion
+      ExecShellWait "open" "$0\uninstall.exe /S"
```

Comment:
Unfortunately this not only seems to not wait, it also seems to fail to uninstall the previous version, so the user ends up with two Anki icons on their desktop, one for the old version and one for the new one.

I experimented with using ExecWait and https://stackoverflow.com/questions/24595887/waiting-for-nsis-uninstaller-to-finish-in-nsis-installer-either-fails-or-the-uni, but didn't have any luck - maybe it's unable to request elevated permissions?

If you can't think of a way to solve this, maybe another approach we could use is to detect the existing registry key, and if found, abort the install, asking the user to uninstall their previous Anki version first? It's a bit more hassle for them, but is better than having multiple copies floating around.

Reply:
Are you sure it was not uninstalled? In my tests it worked. Uninstall is silent, so you will have to check the folders/regkeys.

`ExecWait` does indeed not work, just as you said, it doesn't work with elevated commands. `ExecShellWait` could execute the `uninstall.exe` though. Only the "wait" part didn't work as expected. I think it's because of the UAC prompt, which most probably already returns something, after you press Yes/No, and doesn't wait for the uninstaller to finish. Therefore install of the new anki and uninstalling of the old anki is done in parallel. They do not share any paths/regkeys, so this shouldn't lead to any problems.

But yes, I agree, it's not the cleanest solution. Therefore I suggest the following:

1. if regkey is found
    1. display message with something like "Old version of anki found. Please uninstall it first and execute this installer again. Starting uninstaller now..."
    2. execute uninstaller.exe without silent
    3. Abort installer.
2. User executes installer again the if at `1.` is not executed

This way, we can be sure that the uninstaller finished successfully.

Reply:
Another issue that should be addressed, before this is merged: this change might have an impact on package managers like chocolatey. Are there any others that you are aware of? The maintainers should most probably be informed.

Reply:
I implemented this in 08fd98c4b638346e716826ed03330a5daea37423

I works now like described above. Some possible changes would be:

* call `uninstall.exe` with the `/S` flag, the user would not see the uninstall wizard
* sleep for a certain amount of seconds then check if the uninstall is done (this seems a little bit hacky though)

Reply:
> Are you sure it was not uninstalled? In my tests it worked. Uninstall is silent, so you will have to check the folders/regkeys.

Yep, I tested again with the same results - no removed files/folders and reg entry unchanged. Your latest code seems to work correctly though, which is great. I think it's fine to ask the user to re-run the installer after uninstalling; it's not much trouble, and the alternative solutions are hacky.

> this change might have an impact on package managers like chocolatey

If they're just invoking the official installer, users could presumably work around any issues by removing the package and adding it again I guess?

Reply:
> Yep, I tested again with the same results - no removed files/folders and reg entry unchanged. Your latest code seems to work correctly though, which is great. I think it's fine to ask the user to re-run the installer after uninstalling; it's not much trouble, and the alternative solutions are hacky.

perfect! I commited your suggestion

> If they're just invoking the official installer, users could presumably work around any issues by removing the package and adding it again I guess?

Yes most probably your are right. I will still send them a message to inform them. I myself am a chocolatey user so I have some personal involvement in this ;)